version: '3.8'
name: avora

services:
  web:
    build: .
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - .:/app
    ports:
      - "8000:8000"
    env_file:
      - ./.env
    depends_on:
      - db
      - redis
    restart: on-failure

  db:
    image: pgvector/pgvector:pg16
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    env_file:
      - ./.env
    environment:
      - POSTGRES_PASSWORD=${DB_PASSWORD}
      - POSTGRES_USER=${DB_USER}
      - POSTGRES_DB=${DB_NAME}
    ports:
      - "5432:5432"

  redis:
    image: redis:7
    ports:
      - "6379:6379"
    volumes:
      - redis_data:/data

  celery_worker:
    build: .
    command: celery -A scoring worker -l INFO --concurrency=5
    volumes:
      - .:/app
    env_file:
      - ./.env
    depends_on:
      - web
      - db
      - redis

  celery_beat:
    build: .
    command: celery -A scoring beat -l INFO
    volumes:
      - .:/app
    env_file:
      - ./.env
    depends_on:
      - web
      - db
      - redis

volumes:
  postgres_data:
  redis_data:
